internal/beater/server_test.go

```diff
@@ -487,44 +496,11 @@ func TestServerElasticsearchOutput(t *testing.T) {
 		t.Fatal("timed out waiting for bulk request")
 	}
 
-	snapshot := monitoring.CollectStructSnapshot(monitoring.Default.GetRegistry("libbeat"), monitoring.Full, false)
-	assert.Equal(t, map[string]interface{}{
-		"output": map[string]interface{}{
-			"events": map[string]interface{}{
-				"acked":   int64(0),
-				"active":  int64(5),
-				"batches": int64(0),
-				"failed":  int64(0),
-				"toomany": int64(0),
-				"total":   int64(5),
-			},
-			"type": "elasticsearch",
-			"write": map[string]interface{}{
-				// _bulk requests haven't completed, so bytes flushed won't have been updated.
-				"bytes": int64(0),
-			},
-		},
-		"pipeline": map[string]interface{}{
-			"events": map[string]interface{}{
-				"total": int64(5),
-			},
-		},
-	}, snapshot)
-
-	snapshot = monitoring.CollectStructSnapshot(monitoring.Default.GetRegistry("output"), monitoring.Full, false)
-	assert.Equal(t, map[string]interface{}{
-		"elasticsearch": map[string]interface{}{
-			"bulk_requests": map[string]interface{}{
-				"available": int64(9),
-				"completed": int64(0),
-			},
-			"indexers": map[string]interface{}{
-				"active":    int64(1),
-				"destroyed": int64(0),
-				"created":   int64(0),
-			},
-		},
-	}, snapshot)
+	monitoringtest.ExpectOtelMetrics(t, reader, map[string]any{
```

Comment:
It seems we're missing a bunch of non-zero metrics that we had before, is that expected?

 - `elasticsearch.indexers.active`
 - `output.events.active`
 - `output.events.total`
 - `pipeline.events.total`

Reply:
Ah, that would be expected here since this code is only looking at OTel metrics. We're missing a test in beatcmd for the mapping.

Reply:
Added a unit test in beatcmd